lib/ex_doc/doc_ast.ex

```diff
@@ -170,8 +170,21 @@ defmodule ExDoc.DocAST do
 
   defp highlight_code_block(full_block, pre_attr, lang, code, highlight_info, outer_opts) do
     case pick_language_and_lexer(lang, highlight_info, code) do
-      {_language, nil, _opts} -> full_block
-      {lang, lexer, opts} -> render_code(pre_attr, lang, lexer, opts, code, outer_opts)
+      {_language, nil, _opts} ->
+        full_block
+
+      {lang, lexer, opts} ->
+        try do
+          render_code(pre_attr, lang, lexer, opts, code, outer_opts)
+        rescue
+          MatchError ->
+            IO.puts(:stderr, [
+              IO.ANSI.format([:yellow, "warning: ", :reset]),
+              "Crashed while formatting #{full_block}\n"
+            ])
```

Comment:
Is this for debugging or is this necessary permanently because "invalid" code blocks are being given?

Reply:
I would prefer to always have it as then we can put the style and github will render it with the correct syntax highlighting without having to fix the bug in the specific makeup flavor that crashes. I found that there are quite a few of our C code snippets that fail in `makeup_c`.